src/main/java/uk/gov/register/presentation/AssetsBundleCustomErrorHandler.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.register.presentation;
+
+import io.dropwizard.setup.Environment;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.handler.ErrorHandler;
+import org.glassfish.hk2.api.ServiceLocator;
+import org.glassfish.jersey.servlet.ServletContainer;
+import org.thymeleaf.TemplateEngine;
+import org.thymeleaf.context.WebContext;
+import org.thymeleaf.resourceresolver.FileResourceResolver;
+import org.thymeleaf.templateresolver.TemplateResolver;
+import uk.gov.register.presentation.config.RegistersConfiguration;
+import uk.gov.register.thymeleaf.ThymeleafResourceResolver;
+
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+public class AssetsBundleCustomErrorHandler extends ErrorHandler {
+    private final Environment environment;
+    private final TemplateEngine engine;
+
+    public AssetsBundleCustomErrorHandler(Environment environment) {
+        this.environment = environment;
+
+        TemplateResolver templateResolver = new TemplateResolver();
+        String baseDirForTemplates = System.getProperty("baseDirForTemplates");
+        if (baseDirForTemplates != null) {
+            templateResolver.setResourceResolver(new FileResourceResolver());
+            templateResolver.setPrefix(baseDirForTemplates + "/templates/");
+        } else {
+            templateResolver.setResourceResolver(new ThymeleafResourceResolver());
+            templateResolver.setPrefix("/templates/");
+        }
+        templateResolver.setTemplateMode("HTML5");
+        templateResolver.setCacheable(false);
+        templateResolver.setCharacterEncoding(StandardCharsets.UTF_8.name());
+        engine = new TemplateEngine();
+        engine.setTemplateResolver(templateResolver);
+    }
+
+    @Override
+    public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
which errors does this handle? is it just 404s or all 4xxs or all 4xxs + all 5xxs?